src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -0,0 +1,926 @@
+
+            ExitNow();
+        }
+    }
+
+    // Stop the CHIPoBLE GATT service if needed.
+    if (mServiceMode != ConnectivityManager::kCHIPoBLEServiceMode_Enabled && GetFlag(mFlags, kFlag_GATTServiceStarted))
+    {
+        // TODO: what to do about existing connections??
+    }
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Disabling CHIPoBLE service due to error: %s", ErrorStr(err));
+        mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_Disabled;
+    }
+}
+
+CHIP_ERROR BLEManagerImpl::InitESPBleLayer(void)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(!GetFlag(mFlags, kFlag_ESPBLELayerInitialized), /* */);
+
+    err = esp_nimble_hci_and_controller_init();
+
+    nimble_port_init();
+
+    SetFlag(mFlags, kFlag_ESPBLELayerInitialized);
+
+exit:
+    return err;
+}
+
+CHIP_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
+{
+    CHIP_ERROR err;
+    struct ble_hs_adv_fields fields = { 0 }, scan_rsp = { 0 };
+
+    ESP32ChipServiceData chipServiceData;
+
+    // If a custom device name has not been specified, generate a CHIP-standard name based on the
+    // bottom digits of the Chip device id.
+    const uint16_t discriminator = 0x0F00;
+    if (!GetFlag(mFlags, kFlag_UseCustomDeviceName))
+    {
+        snprintf(mDeviceName, sizeof(mDeviceName), "%s%04" PRIX32, CHIP_DEVICE_CONFIG_BLE_DEVICE_NAME_PREFIX, discriminator);
+        mDeviceName[kMaxDeviceNameLength] = 0;
+    }
+
+    // Configure the BLE device name.
+    err = ble_svc_gap_device_name_set(mDeviceName);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "ble_svc_gap_device_name_set() failed: %s", ErrorStr(err));
+        ExitNow();
+    }
+
+    // Configure the contents of the advertising packet.
+    fields.name             = (uint8_t *) mDeviceName;
+    fields.name_len         = strlen(mDeviceName);
+    fields.name_is_complete = 1;
+
+    fields.tx_pwr_lvl_is_present = 0;
+    fields.svc_data_uuid16       = (uint8_t *) UUID16_CHIPoBLEService;
+    fields.svc_data_uuid16_len   = sizeof(UUID16_CHIPoBLEService);
+    fields.flags                 = BLE_HS_ADV_F_DISC_GEN | BLE_HS_ADV_F_BREDR_UNSUP;
+
+    // Construct the Chip BLE Service Data to be sent in the scan response packet.  On the ESP32,
+    // the buffer given to esp_ble_gap_config_adv_data() must begin with the Chip BLE service UUID.
+    memcpy(chipServiceData.ServiceUUID, (const void *) UUID16_CHIPoBLEService, sizeof(chipServiceData.ServiceUUID));
+    err = ConfigurationMgr().GetBLEDeviceIdentificationInfo(chipServiceData.DeviceIdInfo);
+    SuccessOrExit(err);
+
+    scan_rsp.tx_pwr_lvl_is_present = 1;
+    scan_rsp.tx_pwr_lvl            = BLE_HS_ADV_TX_PWR_LVL_AUTO;
+    scan_rsp.svc_data_uuid16       = (uint8_t *) UUID16_CHIPoBLEService;
+    scan_rsp.svc_data_uuid16_len   = sizeof(UUID16_CHIPoBLEService);
+
+    ble_gap_adv_set_fields(&fields);
+    ble_gap_adv_rsp_set_fields(&scan_rsp);
```

Comment:
I tried to change the advertising data to look likes what the spec says. By any chance can you use raw data here ? That makes it easier to read imho. See https://github.com/project-chip/connectedhomeip/blob/master/src/platform/ESP32/BLEManagerImpl.cpp#L650

Also for the moment rendezvous expects the advertisement data to looks like the given link